src/participant/participant.ts

```diff
@@ -1802,4 +1850,79 @@ Please see our [FAQ](https://www.mongodb.com/docs/generative-ai-faq/) for more i
       .getSavedConnections()
       .map((connection) => connection.name);
   }
+
+  changeExportToLanguageAddons(
+    exportToLanguageAddons: ExportToLanguageAddons
+  ): Promise<boolean> {
+    this._exportToLanguageCodeLensProvider.refresh(exportToLanguageAddons);
+    return this._transpile();
+  }
+
+  async exportToLanguage(language: string): Promise<boolean> {
+    const editor = vscode.window.activeTextEditor;
+
+    if (!isPlayground(editor?.document.uri)) {
+      await vscode.window.showErrorMessage(
+        'Please select one or more lines in the playground.'
+      );
+      return false;
+    }
+
+    const selectedText = getSelectedText();
+    const codeToTranspile = selectedText || getAllText();
+
+    this._exportToLanguageCodeLensProvider.refresh({
+      ...this._exportToLanguageCodeLensProvider._exportToLanguageAddons,
+      codeToTranspile,
+      language,
+    });
+
+    return this._transpile();
+  }
+
+  async _transpile(): Promise<boolean> {
+    const { codeToTranspile, driverSyntax, language } =
+      this._exportToLanguageCodeLensProvider._exportToLanguageAddons;
+
+    log.info(`Exporting to the '${language}' language...`);
+
+    try {
+      const content = await this.exportPlaygroundToLanguage({
+        code: codeToTranspile,
+        language,
+        includeDriverSyntax: driverSyntax,
+      });
+
+      log.info(`The playground was exported to ${language}`, {
+        code: codeToTranspile,
+        language,
+        includeDriverSyntax: driverSyntax,
+      });
+
+      await vscode.commands.executeCommand(
+        EXTENSION_COMMANDS.SHOW_EXPORT_TO_LANGUAGE_RESULTS,
+        { content, language }
```

Comment:
Right now we're getting the full response from the model before we show it. We could stream it to a preview panel. Not something we need to do now, but it could be a nice improvement down the line. Mostly calling it out incase it wasn't something we were looking at before.

Reply:
I was considering this but matched the behaviour to other cases when we open the results pane with all as one ready results without streaming them.